### PR TITLE
fix: implement default method for schema registry client

### PR DESCRIPTION
**Pull request by lihaosky**
### Description 

https://github.com/confluentinc/schema-registry/pull/2674 introduced a new default method `registerWithResponse` which throws exception by default. `SandboxedSchemaRegistryClient.java` should implement it. Otherwise `io.confluent.ksql.rest.server.InsertionIntegrationTest.shouldSupportEvolvingSchemasOnInsert` and `KsqlResourceFunctionalTest.shouldInsertIntoValuesForAvroTopic` fail.

### Testing done 
`io.confluent.ksql.rest.server.InsertionIntegrationTest.shouldSupportEvolvingSchemasOnInsert` and `KsqlResourceFunctionalTest.shouldInsertIntoValuesForAvroTopic` pass after the fix.

### Reviewer checklist
- [ ] Ensure docs are updated if necessary. (eg. if a user visible feature is being added or changed).
- [ ] Ensure relevant issues are linked (description should include text like "Fixes #<issue number>")
- [ ] Do these changes have compatibility implications for rollback? If so, ensure that the ksql [command version](https://github.com/confluentinc/ksql/blob/master/ksqldb-rest-app/src/main/java/io/confluent/ksql/rest/server/computation/Command.java#L41) is bumped.

### CODE REVIEW

**Review by vcrfxia — APPROVED** [review]
Thanks @lihaosky !

**Comment by vcrfxia** [comment]
checkstyle failure:
```
14:25:17  [ERROR] /home/jenkins/workspace/ic_Repo_PR_builder_ksql_PR-10000/ksqldb-engine/src/main/java/io/confluent/ksql/services/SandboxedSchemaRegistryClient.java:102:9: Parameter subject should be final. [FinalParameters]
14:25:17  [ERROR] /home/jenkins/workspace/ic_Repo_PR_builder_ksql_PR-10000/ksqldb-engine/src/main/java/io/confluent/ksql/services/SandboxedSchemaRegistryClient.java:102:25: Parameter schema should be final. [FinalParameters]
14:25:17  [ERROR] /home/jenkins/workspace/ic_Repo_PR_builder_ksql_PR-10000/ksqldb-engine/src/main/java/io/confluent/ksql/services/SandboxedSchemaRegistryClient.java:102:46: Parameter normalize should be final. [FinalParameters]
```

**Review by aliehsaeedii — APPROVED** [review]

File: ksqldb-engine/src/test/java/io/confluent/ksql/services/SandboxedSchemaRegistryClientTest.java [hunk 1]
```diff
@@ -64,6 +64,7 @@ public static Collection<TestCase<SchemaRegistryClient>> getMethodsToTest() {
           .ignore("register", String.class, Schema.class, int.class, int.class)
           .ignore("register", String.class, ParsedSchema.class, int.class, int.class)
           .ignore("getLatestSchemaMetadata", String.class)
+          .ignore("registerWithResponse", String.class, ParsedSchema.class, boolean.class)
```
Comment (vcrfxia) [inline ksqldb-engine/src/test/java/io/confluent/ksql/services/SandboxedSchemaRegistryClientTest.java:67]: Why is this added here? This list is for methods which are _not_ implemented by the proxy.

File: ksqldb-functional-tests/src/test/java/io/confluent/ksql/test/serde/json/ValueSpecJsonSchemaSerdeSupplierTest.java [hunk 2]
```diff
@@ -33,6 +37,8 @@ public void shouldSerializeAndDeserializeDecimalsWithOutStrippingTrailingZeros()
 
     when(srClient.getLatestSchemaMetadata("t-value"))
         .thenReturn(new SchemaMetadata(0, 1, ""));
+    when(srClient.registerWithResponse(anyString(), any(), anyBoolean()))
```
Comment (vcrfxia) [inline ksqldb-functional-tests/src/test/java/io/confluent/ksql/test/serde/json/ValueSpecJsonSchemaSerdeSupplierTest.java:40]: Ideally we'd fix this by switching away from `mock(SchemaRegistryClient)` to `MockSchemaRegistryClient`, the same as in https://github.com/confluentinc/ksql/pull/9982, but since this is for an older branch I don't feel as strongly. The costs to not doing that are (1) inconsistency within the codebase and (2) if another new method is added to SchemaRegistryClient later on, then we'll have to make another analogous change again in the future.

File: ksqldb-engine/src/test/java/io/confluent/ksql/services/SandboxedSchemaRegistryClientTest.java (hunk 1 above)
Comment (lihaosky) [reply ksqldb-engine/src/test/java/io/confluent/ksql/services/SandboxedSchemaRegistryClientTest.java:67]: It was failing before. Looks like this is a list of method implemented by proxy which will be ignored and not throw unsupported error

Comment (lihaosky) [reply ksqldb-engine/src/test/java/io/confluent/ksql/services/SandboxedSchemaRegistryClientTest.java:67]: This time, it's complaining about
```
  default String tenant() {
    return DEFAULT_TENANT;
  }
```

Comment (vcrfxia) [reply ksqldb-engine/src/test/java/io/confluent/ksql/services/SandboxedSchemaRegistryClientTest.java:67]: I see. Turns out we have to add the new method in both places, since our sandbox client provides a custom implementation. Should be fixed now in the latest commit.

**Comment by vcrfxia** [comment]
PR builder has four integration test failures, none of which appear related to changes in this PR. Two are flaky (don't fail when run locally) but the other two are reproducible.

`io.confluent.ksql.rest.integration.SystemAuthenticationFunctionalTest$HttpsNoMutualAuth.shouldHeartbeatSuccessfully` fails with 
```
java.lang.NoSuchMethodError: 'void io.netty.handler.ssl.SniHandler.<init>(io.netty.util.AsyncMapping, long)'
	at io.vertx.core.net.impl.VertxSniHandler.<init>(VertxSniHandler.java:33)
	at io.vertx.core.net.impl.SSLHelper.createSniHandler(SSLHelper.java:347)
	at io.vertx.core.net.impl.SSLHelper.createHandler(SSLHelper.java:352)
	at io.vertx.core.http.impl.HttpServerWorker.configurePipeline(HttpServerWorker.java:139)
	at io.vertx.core.http.impl.HttpServerWorker.handle(HttpServerWorker.java:132)
	at io.vertx.core.http.impl.HttpServerWorker.handle(HttpServerWorker.java:55)
	at io.vertx.core.net.impl.ServerChannelLoadBalancer.initChannel(ServerChannelLoadBalancer.java:61)
	at io.netty.channel.ChannelInitializer.initChannel(ChannelInitializer.java:129)
	at io.netty.channel.ChannelInitializer.handlerAdded(ChannelInitializer.java:112)
	at io.netty.channel.AbstractChannelHandlerContext.callHandlerAdded(AbstractChannelHandlerContext.java:1114)
	at io.netty.channel.DefaultChannelPipeline.callHandlerAdded0(DefaultChannelPipeline.java:609)
	at io.netty.channel.DefaultChannelPipeline.access$100(DefaultChannelPipeline.java:46)
	at io.netty.channel.DefaultChannelPipeline$PendingHandlerAddedTask.execute(DefaultChannelPipeline.java:1463)
	at io.netty.channel.DefaultChannelPipeline.callHandlerAddedForAllHandlers(DefaultChannelPipeline.java:1115)
	at io.netty.channel.DefaultChannelPipeline.invokeHandlerAddedIfNeeded(DefaultChannelPipeline.java:650)
	at io.netty.channel.AbstractChannel$AbstractUnsafe.register0(AbstractChannel.java:514)
	at io.netty.channel.AbstractChannel$AbstractUnsafe.access$200(AbstractChannel.java:429)
	at io.netty.channel.AbstractChannel$AbstractUnsafe$1.run(AbstractChannel.java:486)
	at io.netty.util.concurrent.AbstractEventExecutor.runTask(AbstractEventExecutor.java:174)
	at io.netty.util.concurrent.AbstractEventExecutor.safeExecute(AbstractEventExecutor.java:167)
	at io.netty.util.concurrent.SingleThreadEventExecutor.runAllTasks(SingleThreadEventExecutor.java:470)
	at io.netty.channel.nio.NioEventLoop.run(NioEventLoop.java:569)
	at io.netty.util.concurrent.SingleThreadEventExecutor$4.run(SingleThreadEventExecutor.java:997)
	at io.netty.util.internal.ThreadExecutorMap$2.run(ThreadExecutorMap.java:74)
	at io.netty.util.concurrent.FastThreadLocalRunnable.run(FastThreadLocalRunnable.java:30)
	at java.base/java.lang.Thread.run(Thread.java:829)
```
The test client is unable to reach the ksql server in order to check its status.

`io.confluent.ksql.rest.server.ConnectIntegrationTest` fails to initialize with
```
java.lang.NoClassDefFoundError: org/glassfish/jersey/internal/DynamicFeatureConfigurator
	at org.glassfish.jersey.server.ApplicationHandler.initialize(ApplicationHandler.java:273)
	at org.glassfish.jersey.server.ApplicationHandler.<init>(ApplicationHandler.java:261)
	at org.glassfish.jersey.servlet.WebComponent.<init>(WebComponent.java:311)
	at org.glassfish.jersey.servlet.ServletContainer.init(ServletContainer.java:154)
	at org.glassfish.jersey.servlet.ServletContainer.init(ServletContainer.java:339)
	at javax.servlet.GenericServlet.init(GenericServlet.java:244)
	at org.eclipse.jetty.servlet.ServletHolder.initServlet(ServletHolder.java:632)
	at org.eclipse.jetty.servlet.ServletHolder.initialize(ServletHolder.java:415)
	at org.eclipse.jetty.servlet.ServletHandler.lambda$initialize$0(ServletHandler.java:750)
	at java.base/java.util.stream.SortedOps$SizedRefSortingSink.end(SortedOps.java:357)
	at java.base/java.util.stream.AbstractPipeline.copyInto(AbstractPipeline.java:485)
	at java.base/java.util.stream.AbstractPipeline.wrapAndCopyInto(AbstractPipeline.java:474)
	at java.base/java.util.stream.StreamSpliterators$WrappingSpliterator.forEachRemaining(StreamSpliterators.java:312)
	at java.base/java.util.stream.Streams$ConcatSpliterator.forEachRemaining(Streams.java:735)
	at java.base/java.util.stream.ReferencePipeline$Head.forEach(ReferencePipeline.java:658)
	at org.eclipse.jetty.servlet.ServletHandler.initialize(ServletHandler.java:774)
	at org.eclipse.jetty.servlet.ServletContextHandler.startContext(ServletContextHandler.java:379)
	at org.eclipse.jetty.server.handler.ContextHandler.doStart(ContextHandler.java:916)
	at org.eclipse.jetty.servlet.ServletContextHandler.doStart(ServletContextHandler.java:288)
	at org.eclipse.jetty.util.component.AbstractLifeCycle.start(AbstractLifeCycle.java:73)
	at org.apache.kafka.connect.runtime.rest.RestServer.initializeResources(RestServer.java:295)
	at org.apache.kafka.connect.runtime.Connect.start(Connect.java:54)
	at org.apache.kafka.connect.cli.ConnectDistributed.startConnect(ConnectDistributed.java:149)
	at io.confluent.ksql.rest.server.ConnectExecutable.startAsync(ConnectExecutable.java:58)
	at io.confluent.ksql.rest.server.ConnectIntegrationTest.setUpClass(ConnectIntegrationTest.java:120)
	at java.base/jdk.internal.reflect.NativeMethodAccessorImpl.invoke0(Native Method)
	at java.base/jdk.internal.reflect.NativeMethodAccessorImpl.invoke(NativeMethodAccessorImpl.java:62)
	at java.base/jdk.internal.reflect.DelegatingMethodAccessorImpl.invoke(DelegatingMethodAccessorImpl.java:43)
	at java.base/java.lang.reflect.Method.invoke(Method.java:566)
	at org.junit.runners.model.FrameworkMethod$1.runReflectiveCall(FrameworkMethod.java:59)
	at org.junit.internal.runners.model.ReflectiveCallable.run(ReflectiveCallable.java:12)
	at org.junit.runners.model.FrameworkMethod.invokeExplosively(FrameworkMethod.java:56)
	at org.junit.internal.runners.statements.RunBefores.invokeMethod(RunBefores.java:33)
	at org.junit.internal.runners.statements.RunBefores.evaluate(RunBefores.java:24)
	at org.junit.internal.runners.statements.RunAfters.evaluate(RunAfters.java:27)
	at org.junit.rules.ExternalResource$1.evaluate(ExternalResource.java:54)
	at org.junit.rules.ExternalResource$1.evaluate(ExternalResource.java:54)
	at io.confluent.ksql.integration.Retry$1.evaluate(Retry.java:129)
	at org.junit.rules.RunRules.evaluate(RunRules.java:20)
	at org.junit.rules.RunRules.evaluate(RunRules.java:20)
	at org.junit.runners.ParentRunner$3.evaluate(ParentRunner.java:306)
	at org.junit.runners.ParentRunner.run(ParentRunner.java:413)
	at org.junit.runner.JUnitCore.run(JUnitCore.java:137)
	at org.junit.runner.JUnitCore.run(JUnitCore.java:115)
	at org.junit.vintage.engine.execution.RunnerExecutor.execute(RunnerExecutor.java:42)
	at org.junit.vintage.engine.VintageTestEngine.executeAllChildren(VintageTestEngine.java:80)
	at org.junit.vintage.engine.VintageTestEngine.execute(VintageTestEngine.java:72)
	at org.junit.platform.launcher.core.EngineExecutionOrchestrator.execute(EngineExecutionOrchestrator.java:107)
	at org.junit.platform.launcher.core.EngineExecutionOrchestrator.execute(EngineExecutionOrchestrator.java:88)
	at org.junit.platform.launcher.core.EngineExecutionOrchestrator.lambda$execute$0(EngineExecutionOrchestrator.java:54)
	at org.junit.platform.launcher.core.EngineExecutionOrchestrator.withInterceptedStreams(EngineExecutionOrchestrator.java:67)
	at org.junit.platform.launcher.core.EngineExecutionOrchestrator.execute(EngineExecutionOrchestrator.java:52)
	at org.junit.platform.launcher.core.DefaultLauncher.execute(DefaultLauncher.java:114)
	at org.junit.platform.launcher.core.DefaultLauncher.execute(DefaultLauncher.java:86)
	at org.junit.platform.launcher.core.DefaultLauncherSession$DelegatingLauncher.execute(DefaultLauncherSession.java:86)
	at org.apache.maven.surefire.junitplatform.LazyLauncher.execute(LazyLauncher.java:55)
	at org.apache.maven.surefire.junitplatform.JUnitPlatformProvider.execute(JUnitPlatformProvider.java:223)
	at org.apache.maven.surefire.junitplatform.JUnitPlatformProvider.invokeAllTests(JUnitPlatformProvider.java:175)
	at org.apache.maven.surefire.junitplatform.JUnitPlatformProvider.invoke(JUnitPlatformProvider.java:139)
	at org.apache.maven.surefire.booter.ForkedBooter.runSuitesInProcess(ForkedBooter.java:456)
	at org.apache.maven.surefire.booter.ForkedBooter.execute(ForkedBooter.java:169)
	at org.apache.maven.surefire.booter.ForkedBooter.run(ForkedBooter.java:595)
	at org.apache.maven.surefire.booter.ForkedBooter.main(ForkedBooter.java:581)
Caused by: java.lang.ClassNotFoundException: org.glassfish.jersey.internal.DynamicFeatureConfigurator
	at java.base/jdk.internal.loader.BuiltinClassLoader.loadClass(BuiltinClassLoader.java:581)
	at java.base/jdk.internal.loader.ClassLoaders$AppClassLoader.loadClass(ClassLoaders.java:178)
	at java.base/java.lang.ClassLoader.loadClass(ClassLoader.java:522)
	... 63 more
```
when trying to start the connect executable.
